TACC-Cloud/geoapi
# Task/DES-1672 handle v1 geojson with null coordinates

## Overview: ##

Handle Hazmapper v1 geojson with null coordinates by using `geopandas` instead of `geojson`.  

Note the problem is when `geojson` checks the coordinates here: https://github.com/jazzband/geojson/blob/master/geojson/geometry.py#L48-L56 
it doesn't like that they are `null` (see this: https://github.com/jazzband/geojson/blob/master/geojson/geometry.py#L8-L12)

## PR Status: ##


## Related Jira tickets: ##

* [DES-1672](https://jira.tacc.utexas.edu/browse/DES-1672)

## Summary of Changes: ##
0) Used `geopandas` instead of `geojson`
1) Added unit tests for properties
2) Added unit tests for v1 files that had null values
3) Added unit tests for v1 files that have properties containing lists

## Testing Steps: ##
1. Load  `/PRJ-2440/Brandenberg/July5/Brandenberg_July_5_2019.geojson`
2. Confirm everything in properties is there.

## Notes: ##

TODO (need to fix 2 problems; so maybe GeoPandas/Fioina isn't the way to go):
- [ ]  Using GeoPandas/Fiona (GDAL/ogr) does not support  tuple or list properties (need to support `geoapi/tests/fixtures/hazmapperv1_with_list_properties.json`)
- [ ]  GeoPandas/Fiona (GDAL/ogr) has just one type for each "column" of data so if we have two features with varying properties (like feature 1 `{"prop1": 1.0}` and feature 2 `{"prop1": "one"}`, then the feature one property gets changed to `"1.0"` instead of `1.0`.  do we need to support this?
